tests/testsuite.src/syn_misc.at

```diff
@@ -8348,3 +8348,253 @@ AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [1])
+       END PROGRAM prog5.
+])
+
+# There is no reason for this one to fail, it has the RECURSIVE flag...
+# but no IDENTIFICATION DIVISION before the second PROGRAM-ID
+
+AT_CHECK([$COMPILE prog5.cob], [1], [],
+[prog5.cob:18: error: syntax error, unexpected Identifier
+])
+AT_DATA([prog5.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog5 RECURSIVE.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 counter PIC 9(1).
+       PROCEDURE DIVISION.
+          DISPLAY "HELLO WORLD".
+          CALL "entry".
+          GOBACK.
+          ENTRY "entry".
+	  ADD 1 TO counter.
+	  DISPLAY "entry " counter.
+	  IF counter < 5 THEN
+            CALL "inside-program"
+          END-IF.
+	  GOBACK.
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. inside-program RECURSIVE.
+       PROCEDURE DIVISION.
+           CALL "entry".
+       END PROGRAM inside-program.
+       END PROGRAM prog5.
+])
+
+# Adding the IDENTIFICATION DIVISION makes it work... why ?
+
+AT_CHECK([$COMPILE prog5.cob])
+
+AT_CHECK([$COBCRUN_DIRECT ./prog5], [0], [HELLO WORLD
+entry 1
+entry 2
+entry 3
+entry 4
+entry 5
+])
+
+AT_DATA([prog6.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog6 RECURSIVE.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 counter PIC 9(1).
+       PROCEDURE DIVISION.
+          DISPLAY "HELLO WORLD".
+          CALL "entry".
+          GOBACK.
+          ENTRY "entry".
+	  ADD 1 TO counter.
+	  DISPLAY "entry " counter.
+	  IF counter < 5 THEN
+            CALL "inside-program"
+          END-IF.
+	  GOBACK.
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. inside-program.
+       DATA DIVISION.
+       LOCAL-STORAGE SECTION.
+       PROCEDURE DIVISION.
+           CALL "entry".
+       END PROGRAM inside-program.
+       END PROGRAM prog6.
+])
+
+# No LOCAL-STORAGE in nested programs ? Microfocus documentation does
+# not display such a limitation
+
+AT_CHECK([$COMPILE prog6.cob], [1], [],
+[prog6.cob:21: error: LOCAL-STORAGE not allowed in nested programs
+])
+
+AT_DATA([prog6.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog6.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 counter PIC 9(1).
+       LOCAL-STORAGE SECTION.
+       PROCEDURE DIVISION.
+          DISPLAY "HELLO WORLD".
+          CALL "entry".
+          GOBACK.
+          ENTRY "entry".
+	  ADD 1 TO counter.
+	  DISPLAY "entry " counter.
+	  IF counter < 5 THEN
+            CALL "inside-program"
+          END-IF.
+	  GOBACK.
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. inside-program RECURSIVE.
+       PROCEDURE DIVISION.
+           CALL "entry".
+       END PROGRAM inside-program.
+       END PROGRAM prog6.
+])
+
+# This is ok to run if LOCAL-STORAGE implies RECURSIVE. Note that the
+# need for RECURSIVE on a nested program that is currently not allowed
+# to have a LOCAL-STORAGE section is paradoxal.
+
+AT_CHECK([$COMPILE prog6.cob])
+
+# Oups, this one segfaults
```

Comment:
That should not sigsegv, should it?
Again this may be moved to a separate test that expects a zero return including an `AT_XFAIL`.

Note that per standard `prog6` should have the recursive attribute (only MF implies recursive with local-storage) and if it isn't there (which we should have a test above) then it should nicely abort on the first call of the entry point.

Reply:
Apparently this no longer segfaults (I removed the AT_XFAIL).

Reply:
let's just add a `RECURSIVE` to the program then and remove the confusing comments here, then consider this part done, too.

Reply:
I'm not sur what was Emilien's intentions behind this test.
There are two variants of this "prog6", one where the main program is `RECURSIVE` and the nested program has a `LOCAL-STORAGE`, and another one where the main program has a `LOCAL-STORAGE` and the nested program is `RECURSIVE`.

Reply:
Possibly a codegen error that occurred during development?

_If we don't have such tests yet_ then we should have two - with the `RECURSIVE` attribute in both (expected test run) and one program with the called one not being `RECURSIVE` (should already create a nice abort) and a separate one with the outer not being `RECURSIVE` (which seems to be an XFAIL not aborting currently) - all in run_misc.at.

... and another one in syn_misc (XFAIL) that tests for the compiler aborting on all of the non-recursive programs being called recursive (= as nested program).